lib/chef/mixin/powershell_exec.rb

```diff
@@ -0,0 +1,39 @@
+#
+# Author:: Stuart Preston (<stuart@chef.io>)
+# Copyright:: Copyright 2018, Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "chef/powershell"
+
+class Chef
+  module Mixin
+    module PowershellExec
+      # Run a command under PowerShell via a managed (.NET) COM interop API.
+      # This implementation requires the managed dll to be registered on the
+      # target machine.
+      #
+      # Requires: .NET Framework 4.0 or higher on the target machine.
+      #
+      # @param script [String] script to run
+      # @return [Chef::PowerShell] output
+      def powershell_exec(*command_args)
+        script = command_args.first
+        options = command_args.last.is_a?(Hash) ? command_args.last : nil
```

Comment:
This should probably just be `powershell_exec(script)` for now.  I don't think options are being used at all here now and can safely be introduced later when there's actually options to use (and `**options` should get used rather than the `last.is_a?(Hash)` stuff which is probably some ruby 1.8 kludginess in shell_out which we've not yet dared to do away with.  The array style passing of arguments on windows is also busted (i tried adding windows support once to bake the poise_shell_out helper directly into mixlib-shellout, but its complicated and brittle -- https://blogs.msdn.microsoft.com/twistylittlepassagesallalike/2011/04/23/everyone-quotes-command-line-arguments-the-wrong-way/).

Reply:
Yeah, I was following powershell_out's style of doing things here.  I'll revert back to taking the single argument for now :)